File: reconstructQueue/reconstructQueueSoluation.java
package reconstructQueue;
import java.util.*;

class reconstructQueueSoluation {
   
    // the solution for reconstruct the Queue
    public  ArrayList<int[]> reconstructQueue(int[][] people) {
   
        // This to sorted the Queue  based on the height decresing order
        int[][] hightSortedPeople= SortQueueBasedOnHight(people);
        // This to add people at the index , picking the max height first
        ArrayList<int[]> result=maxHeightFirst( hightSortedPeople) ;

        return result;
    }

    // method to sorted based on hight
    public  int[][] SortQueueBasedOnHight(int[][] people) {

        Arrays.sort(people, (a,b)->{
             if(a[0]==b[0]) return a[1]-b[1];
             return b[0]-a[0];
        });
 
        return people;
 
    }
    
    // method to picking the max height first
    public  ArrayList<int[]> maxHeightFirst(int[][] hightSortedPeople) {

        ArrayList<int[]> res= new ArrayList<>();
        for(int[] person: hightSortedPeople){
            res.add(person[1], person);
        }
        return res;
    }


    

}

File: reconstructQueue/reconstructQueueSoluation.php

<?php

class reconstructQueueSoluation {
   
    // the solution for reconstruct the Queue
    public function reconstructQueue( $people) {
        // This to sorted the Queue  based on the height decresing order
        $hightSortedPeople= $this->SortQueueBasedOnHight($people);
        // This to add people at the index , picking the max height first
        $result= $this->maxHeightFirst($hightSortedPeople) ;
        return $result;
    }

    // method to sorted based on hight
    private function SortQueueBasedOnHight($people) {

        usort($people, function ($a, $b) {
            if ($a[0] == $b[0]) {
                return $a[1]-$b[1];
            }
            else{
                return  $b[0]-$a[0];
            }
        });
        return $people;
    }
    
    // method to picking the max height first
    private  function maxHeightFirst($hightSortedPeople) {

        $res= [];
        foreach($hightSortedPeople as $person){
            array_splice($res,$person[1],0,array($person));
        }
      
        return $res;
    }

}


// test case should output [[4,0],[5,0],[2,2],[3,2],[1,4],[6,0]]
$people= [[6,0],[5,0],[4,0],[3,2],[2,2],[1,4]];
$class=new reconstructQueueSoluation();
print_r($class->reconstructQueue($people));